odoo/models.py

```diff
@@ -1601,6 +1601,55 @@ def name_create(self, name):
             _logger.warning("Cannot execute name_create, no _rec_name defined on %s", self._name)
             return False
 
+    @api.model
+    def complete_field(self, field, key, field_domain=None, parent_domain=None, operator='ilike', limit=9):
+        """ Completes the m2o field ``field`` to match the provided value ``key``.
+        Only returns record references linked from a record of the current model.
+
+        :param str field: field to complete
+        :param str key: ilike value for the field to complete
+        :param field_domain: filter to apply on acceptable ``field`` records
+        :param parent_domain: filter to apply on current model records
+        :param str operator: matching operator between ``field`` and ``key``, defaults to ``ilike``
+        :param int limit: number of results to return, defaults to 9
+        :returns: name_get like results, list of (id, name)
+        """
+        field_info = self._fields[field]
+        if field_info.type != 'many2one':
+            raise ValueError("Completed field must be a many2one, got %s" % field_info.type)
+
+        domain = [(field, operator, key)]
+        if field_domain:
+            for section in field_domain:
+                if isinstance(section, str):
+                    domain.append(section)
+                else:
+                    # prefix each field of the section with the name of the
+                    # field: (foo, =, 3) => ($field.foo, =, 3) this way field's
+                    # own values can be filtered from parent
+                    f, o, v = section
```

Comment:
Hi, can variable name be more descriptive??
`field_name, operator_name, value_name = section`
 
instead of 

`f, o, v = section`